### PR TITLE
OMERO.Matlab: add examples for adding and retrieving transforms from shape objects

### PR DESCRIPTION
**Pull request by bramalingam**
# What this PR does

Adds examples to adding and retrieving transforms from shape objects

# Testing this PR

1. Check if the examples are passing,
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-training/
@sbesson @jburel 

# Related reading

Documentation PR to be opened separately to expose the ellipse example in the Matlab bindings page.

1. background for understanding this PR
http://blog.openmicroscopy.org/data-model/future-plans/2016/06/20/shape-transforms/
https://uk.mathworks.com/discovery/affine-transformation.html

### CODE REVIEW

**Comment by will-moore** [comment]
This seems to be causing Python Training tests to fail because the Python code is picking up shapes you've created and it assumes they have theZ and theT set: https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-breaking-training/415/console
Python code needs to be updated to import ```unwrap``` from omero.rtypes and make this change everywhere:
```
-     shape['theT'] = s.getTheT().getValue()
+     shape['theT'] = unwrap(s.getTheT())
```

File: examples/Training/matlab/ROIs.m [hunk 1]
```diff
@@ -173,6 +173,124 @@
     roi = session.getUpdateService().saveAndReturnObject(roi);
     fprintf(1, 'Created ROI %g\n', roi.getId().getValue());
 
+    % Create and apply transforms to shapes and add them to ROI
+    
+    % Apply shearing alone to a rectangle (Shear factor set to 10 in the x
+    % and y dimensions in this example)
+    % create rectangle (shape object)
+    rectangle = createRectangle(0, 0, 10, 20);
+    rectangle = setShapeCoordinates(rectangle, 0, 0, 0);
```
Comment (bramalingam) [inline examples/Training/matlab/ROIs.m:182]: @will-moore : setShapeCoordinates sets the Z,C and T for the shapes. And it is setting them to 0,0,0 in this case!

File: examples/Training/matlab/ROIs.m [hunk 2]
```diff
@@ -173,6 +173,124 @@
...
+    newTform.setA02(rdouble(xTranslate));
+    newTform.setA12(rdouble(yTranslate));
+    % apply transform
+    point.setTransform(newTform);
+    
+    % Apply a scale change alone to a polygon object (set scale factor to 10
+    % in the x  and y direction)
+    % create polygon (shape object)
+    polygon = createPolygon([1 5 10 8], [1 5 5 10]);
+    setShapeCoordinates(point, 0, 0, 0);
```
Comment (will-moore) [inline examples/Training/matlab/ROIs.m]: @bramalingam You are not setting coordinates on the new polygon here (copy & paste error 'point').

**Comment by bramalingam** [comment]
@will-moore : Yup found the issue and have pushed a fix.!

**Comment by will-moore** [comment]
Great - so that should fix the tests. I can update the Python training in another PR.

**Comment by jburel** [comment]
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-training/525/ is green
The python training example to be adjusted in another PR

**Pull request merged**